lib/kitto/router.ex

```diff
@@ -32,11 +32,15 @@ defmodule Kitto.Router do
   end
 
   post "widgets/:id" do
-    {:ok, body, conn} = read_body(conn)
+    if Application.get_env(:kitto, :auth_token) && !authenticate(conn) do
```

Comment:
I suggest changing this part to:

```elixir
if authentication_required && !authenticated?(conn) do
```

With `authentication_required` defined as:

```elixir
  defp authentication_required?, do: !!auth_token
```